src/jenkins/ReleaseCandidateStatus.groovy

```diff
@@ -71,43 +73,19 @@ class ReleaseCandidateStatus {
             ]
 
         if (rcNumber != null) {
-            queryMap = [
-                _source: "distribution_build_number",
-                sort: [
-                    [
-                        distribution_build_number: [
-                            order: "desc"
-                        ]
+            queryMap.sort[0].remove("rc_number")
+            queryMap.query.bool.filter.add([
+                    match_phrase: [
+                            rc_number: "${rcNumber}"
```

Comment:
What is this trying to do?

Reply:
Refactored the code to easy logic. See the diff. It does the same. If rc_number is passed the query is different than the default one

Reply:
I guess I didnt get why you need to `            queryMap.sort[0].remove("rc_number")`.

Reply:
Looks like diff is bit messed up. Basically instead of reassigning the `queryMap` here: https://github.com/opensearch-project/opensearch-build-libraries/blob/main/src/jenkins/ReleaseCandidateStatus.groovy#L73-L111
just using a good logic to add and remove the required fields before it

Reply:
Got it now, you are just removing from sort and adding in query.

Thanks.